superbench/benchmarks/micro_benchmarks/ib_loopback_performance.py

```diff
@@ -0,0 +1,218 @@
+from superbench.common.utils import network
+from superbench.benchmarks import BenchmarkRegistry, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class IBLoopbackBenchmark(MicroBenchmarkWithInvoke):
+    """The IB loopback performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'run_perftest_loopback'
+        self.__support_ib_commands = {'write': 'ib_write_bw', 'read': 'ib_read_bw', 'send': 'ib_send_bw'}
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--ib_index',
+            type=int,
+            default=0,
+            required=False,
+            help='The index of ib device.',
+        )
+        self._parser.add_argument(
+            '--n',
+            type=int,
+            default=20000,
+            required=False,
+            help='The iterations of running ib command',
+        )
+        self._parser.add_argument(
+            '--size',
+            type=int,
+            default=8388608,
+            required=False,
+            help='The message size of running ib command. E.g. 8388608.',
+        )
+        self._parser.add_argument(
+            '--commands',
+            type=str,
+            nargs='+',
+            default='write',
+            help='The ib command used to run. E.g. {}.'.format(' '.join(list(self.__support_ib_commands.keys()))),
+        )
+        self._parser.add_argument(
+            '--mode',
+            type=str,
+            default='AF',
+            help='The mode used to run ib command. Eg, AF(all message size) or S(single message size)',
+        )
+        self._parser.add_argument(
+            '--numa',
+            type=int,
+            default=0,
+            required=False,
+            help='The index of numa node.',
+        )
+        self._parser.add_argument(
+            '--x',
+            type=int,
+            default=0,
+            required=False,
+            help='Test uses GID with GID index taken from command.',
+        )
+
+    def __get_numa_cores(self, numa_index):
+        """Get the last two cores from different physical cpu core of NUMA<numa_index>.
+
+        Args:
+            numa_index (int): the index of numa node.
+
+        Return:
+            The last two cores from different physical cpu core of NUMA<numa_index>.
+        """
+        command = 'numactl --hardware | grep "node {} cpus:"'.format(numa_index)
+        output = subprocess.run(
+            command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True, check=False, universal_newlines=True
+        )
+        return output.stdout.splitlines()[0].split(' ')
```

Comment:
read from file directly? need to handle when the numa index is out of range and does not exist
```suggestion
        with Path(f'/sys/devices/system/node/node{numa_index}/cpulist').open('r') as f:
            start, end = f.read().strip().split('-')
            ...
```

Reply:
ok, thanks